axelor-base/src/main/java/com/axelor/apps/base/service/administration/SequenceService.java

```diff
@@ -182,13 +190,40 @@ public String getSequenceNumber(Sequence sequence, LocalDate refDate) {
             .getSingleResult();
     SequenceVersion sequenceVersion = getVersion(seq, refDate);
     String nextSeq = computeNextSeq(sequenceVersion, seq, refDate);
+
+    if (appBaseService.getAppBase().getCheckExistingSequenceOnGeneration()) {
+      this.isSequenceAlreadyExisting(objectClass, fieldName, nextSeq);
+    }
+
     sequenceVersion.setNextNum(sequenceVersion.getNextNum() + seq.getToBeAdded());
     if (sequenceVersion.getId() == null) {
       sequenceVersionRepository.save(sequenceVersion);
     }
     return nextSeq;
   }
 
+  protected void isSequenceAlreadyExisting(Class objectClass, String fieldName, String nextSeq)
+      throws AxelorException {
+    String table = objectClass.getSimpleName();
+    boolean isSequenceAlreadyExisting =
+        CollectionUtils.isNotEmpty(
+            JPA.em()
+                .createQuery(
```

Comment:
You use createQuery so fieldName should be the field's Java name and not SQL name. 
Example, in SyncContactService, you set fieldName to "partner_seq", it should be "partnerSeq"